### PR TITLE
Fix golint warnings

### PR DESCRIPTION
**Pull request by srderson**
## Description

This PR fixes numerous `golint` warnings. There are no functional changes.
## Motivation and Context

Developers are now being asked to run `golint` on all proposed changes and resolve any warnings. This change fixes pre-existing warnings.
## How Has This Been Tested?

I have run `make checks`. There are no new tests as this PR only resolves `golint` warnings and does not add any new functionality.
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Sheehan Anderson sheehan@us.ibm.com

### CODE REVIEW

File: core/ledger/statemgmt/trie/trie_node.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func (trieNode *trieNode) marshal() ([]byte, error) {
 	buffer := proto.NewBuffer([]byte{})
 
 	// write value marker explicitly because rocksdb apis convertes a nil into an empty array and protobuf does it other-way around
-	var valueMarker uint64 = 0
+	var valueMarker uint64 // default value is 0
```
Comment (corecode) [inline core/ledger/statemgmt/trie/trie_node.go]: Now this is a lint warning that I would have ignored.  The assignment is more clear than the comment.

Comment (ghaskins) [inline core/ledger/statemgmt/trie/trie_node.go]: I agree with @corecode

**Comment by christo4ferris** [comment]
I agree, and think that a comment noting the lint exception should be added so we don't get this multiple times

**Comment by srderson** [comment]
@corecode @ghaskins Agree with comments about dropping '= 0' making the assignment less clear. I have reverted the two cases where I did that and added a comment about ignoring the warning per @christo4ferris's suggestion. Thanks.

**Comment by ghaskins** [comment]
LGTM

Reviewed-by: Greg Haskins gregory.haskins@gmail.com

**Pull request merged**